### PR TITLE
dts: npcx: add device-tree files of npcx9 series.

### PR DESCRIPTION
**Pull request by MulinChao**
This PR can be separated into three CLs to introduce NPCX9 series.
1. Replaces series-prefix "npcx7-" with family-"npcx-" for npcx dts nodes such as 'espi-vws-map' and 'miwus-int-map'. 
2. Arrange device-tree files of npcx7 series by following steps:
    Move device-tree declarations of npcx family to npcx.dtsi.
    Leave specific device declarations of npcx7 series to npcx7.dtsi.
    Move chip series related mapping tables such as npcx7-miwus-wui-map.dtsi and so on to npcx/npcx7 folder.
    Move common device-tree declarations used in each npcx series to npcx-miwus-wui-map.dtsi and so on to npcx folder..
3. Add device-tree files of npcx9 series.

Then, the folder tree of nuvoton npcx family will be:
```
dts/arm/nuvoton
        |--npcx
        |    |--npcx7
        |    |    |--npcx7-miwus-wui-map.dtsi
        |    |    |--npcx7-alts-map.dtsi
        |    |    |--.....
        |    +--npcx9
        |    |    |--npcx9-miwus-wui-map.dtsi
        |    |    |--npcx9-alts-map.dtsi
        |    |    +--.....
        |    |--npcx-miwus-wui-map.dtsi
        |    |--npcx-alts-map.dtsi
        |    |--.....
        |
        |--npcx.dtsi
        |--npcx7.dtsi
        |--npcx7m6fb.dtsi
        |--npcx9.dtsi
        +--npcx9m3f.dtsi  
```

### CODE REVIEW

**Review by sjg20 — APPROVED** [review]

File: dts/arm/nuvoton/npcx/npcx7/npcx7-alts-map.dtsi [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
```
Comment (yperess) [inline dts/arm/nuvoton/npcx/npcx7/npcx7-alts-map.dtsi:2]: nit: 2021

File: dts/arm/nuvoton/npcx/npcx7/npcx7-espi-vws-map.dtsi [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
```
Comment (yperess) [inline dts/arm/nuvoton/npcx/npcx7/npcx7-espi-vws-map.dtsi:2]: nit: 2021

File: dts/arm/nuvoton/npcx/npcx7/npcx7-miwus-int-map.dtsi [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
```
Comment (yperess) [inline dts/arm/nuvoton/npcx/npcx7/npcx7-miwus-int-map.dtsi:2]: nit: 2021

File: dts/arm/nuvoton/npcx/npcx7/npcx7-miwus-wui-map.dtsi [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
```
Comment (yperess) [inline dts/arm/nuvoton/npcx/npcx7/npcx7-miwus-wui-map.dtsi:2]: nit: 2021

File: dts/arm/nuvoton/npcx/npcx7/npcx7-alts-map.dtsi (hunk 1 above)
Comment (MulinChao) [reply dts/arm/nuvoton/npcx/npcx7/npcx7-alts-map.dtsi:2]: Hi @yperess:
The npcx7 dts files such as npcx7-alts-map.dtsi, npcx7-espi-vws-map.dtsi and so on are moved from npcx to npcx/npcx7 folder. Do I need to treat them as new files?

Comment (yperess) [reply dts/arm/nuvoton/npcx/npcx7/npcx7-alts-map.dtsi:2]: Ah, my mistake, the PR layout makes it unclear. It looked like they were new files added. Ignore.

**Review by yperess — APPROVED** [review]
LGTM

**Comment by MulinChao** [comment]
> LGTM

I set the other comments as "resolved". Thanks for reviewing.

File: dts/arm/nuvoton/npcx/npcx-alts-map.dtsi [hunk 5]
```diff
@@ -244,21 +235,12 @@
 		alta_32k_out_sl: alta2 {
 			alts = <&scfg 0x0A 0x2 0>;
 		};
-		alta_32kclkin_sl: alta3 {
-			alts = <&scfg 0x0A 0x3 0>;
-		};
```
[the remark this answers is not in the document]
Comment (MulinChao) [reply dts/arm/nuvoton/npcx/npcx-alts-map.dtsi:249]: Yes, bit 3 in DEVALTA is deprecated in npcx7 and later series. Only npcx5 series support this bit.

File: dts/arm/nuvoton/npcx/npcx9/npcx9-alts-map.dtsi [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+		};
+		altg_vcc1_rst_pud_lk: altg5 {
+			alts = <&scfg 0x10 0x5 0>;
+		};
+		altg_no_psl_out_sl: altg6-inv {
+			alts = <&scfg 0x10 0x6 1>;
+		};
+		altg_no_psl_gpo_sl: altg7-inv {
+			alts = <&scfg 0x10 0x7 1>;
+		};
```
[the remark this answers is not in the document]
Comment (MulinChao) [reply dts/arm/nuvoton/npcx/npcx9/npcx9-alts-map.dtsi]: Here is the naming rule for pin-mux phandle in DEVALT registers. 
alt+"DEDVALT group number" + "_no"or "_n" (optional)+ "bit name" 

If the polarity of pin-mux selection is inverse, we will add a "_no" prefix. I made a mistake here. They should be "altg_psl_out_sl" and "altg_psl_gpo_sl" with inverse property "0". Thanks for the suggestions.

[the remark this answers is not in the document]
Comment (MulinChao) [reply dts/arm/nuvoton/npcx/npcx9/npcx9-alts-map.dtsi]: Done.

[the remark this answers is not in the document]
Comment (MulinChao) [reply dts/arm/nuvoton/npcx/npcx9/npcx9-alts-map.dtsi]: Please see above.

**Comment by MulinChao** [comment]
Hi @keith-zephyr:

I have submitted the PR for your request. Any further comment is appreciated. Thank you.

**Review by keith-zephyr — APPROVED** [review]

**Comment by MulinChao** [comment]
Any suggestion for this PR? Or could we merge it since it got two approvals? Thanks.

**Review by yperess — CHANGES_REQUESTED** [review]
One more request. When I tested this code I ran into issues with `ecst.py`. Could you also add the appropriate chips to `CHIPS_INFO` in `ecst_args.py`?

**Comment by MulinChao** [comment]
> One more request. When I tested this code I ran into issues with `ecst.py`. Could you also add the appropriate chips to `CHIPS_INFO` in `ecst_args.py`?

Nuvoton split npcx9 driver implementation into three PRs. The related ecst parameters for npcx9m3f and npcx9m6f are located in the [first CL](https://github.com/zephyrproject-rtos/zephyr/pull/35440/commits/002eee05f029d17d5471c01f7f8a5d019b47bede) of [PR 35440](https://github.com/zephyrproject-rtos/zephyr/pull/35440). If you still encounter issues after applying this, could you give me the build log? Thanks.

**Review by yperess — APPROVED** [review]

**Pull request merged**